frontend/src/components/AddBike/AddBike.tsx

```diff
@@ -1,24 +1,27 @@
-import React, {ChangeEvent, FormEvent, useState} from "react";
-type inputProps = {
-    handleSubmit(title: string): Promise<void>
+import React, { ChangeEvent, FormEvent, useState } from 'react'
+
+type AddBikeProps = {
+   addBike: (newBikeTitle: string) => void
 }
 
-export default function AddBike(props: inputProps){
-    const [inputTitle, setTitle] = useState<string>("")
+export default function AddBike(props: AddBikeProps) {
+   const [inputTitle, setInputTitle] = useState<string>('')
+   const resetInputs = () => setInputTitle('')
 
-    function handleSubmit(event: FormEvent<HTMLFormElement>) {
-        props.handleSubmit(inputTitle)
-            .then(() =>setTitle(""))
+   function handleSubmit(event: FormEvent<HTMLFormElement>) {
+      event.preventDefault()
+      props.addBike(inputTitle)
+      resetInputs()
```

Comment:
Can this be done by directly calling setInputTitle?

Reply:
Yes. :) It can. But I have decided to encapsulate it because, we only have one input field now, but later we may want to adjust the model for multiple properties of a bike and will have a number of inputs which may be then reset after submit with this single function call.